src/main/java/cws/k8s/scheduler/scheduler/nodeassign/LabelAssign.java

```diff
@@ -0,0 +1,82 @@
+package cws.k8s.scheduler.scheduler.nodeassign;
+
+import cws.k8s.scheduler.model.NodeWithAlloc;
+import cws.k8s.scheduler.model.PodWithAge;
+import cws.k8s.scheduler.model.Requirements;
+import cws.k8s.scheduler.model.Task;
+import cws.k8s.scheduler.model.SchedulerConfig;
+import cws.k8s.scheduler.util.NodeTaskAlignment;
+import lombok.extern.slf4j.Slf4j;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.*;
+
+@Slf4j
+public class LabelAssign extends NodeAssign {
+
+    final SchedulerConfig config;
+
+    public LabelAssign( 
+                        final SchedulerConfig config
+                     ){
+        this.config = config;
+    }
+    
+    @Override
+    public List<NodeTaskAlignment> getTaskNodeAlignment( List<Task> unscheduledTasks, Map<NodeWithAlloc, Requirements> availableByNode ) {
+        
+        // get the node-label map 
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String,String> nodelabel = objectMapper.convertValue(config.additional.get("tasklabelconfig"),Map.class);
```

Comment:
Can you force to create a map Map<String,String>. Doesn't it create a Map<Object,Object> here?

Reply:
So this will fail with a Map containing Objects. Please write a test case.